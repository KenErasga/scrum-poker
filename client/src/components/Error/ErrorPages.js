import React from 'react';

export const Page500 = () => <h1>500: Internal Server Error</h1>;
export const Page400 = () => <h1>400: Bad Request</h1>;