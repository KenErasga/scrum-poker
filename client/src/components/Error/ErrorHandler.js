import { useHistory } from 'react-router-dom';
import React, { useState, useEffect, useMemo, createContext, useContext } from 'react';

import { Page400 } from './ErrorPages';

const ErrorHandlerContext = createContext();

const ErrorHandler = props => {
    const [errorMessage, setErrorMessage] = useState();

    const history = useHistory();

    useEffect(() => {
        return history.listen(() => setErrorMessage(undefined));
    }, []);

    const renderContent = () => {
        if (errorMessage === "user-join-failed" ||
            errorMessage === "estimates-update-failed" ||
            errorMessage === "user-has-no-room" ||
            errorMessage === "expand-update-failed" ||
            errorMessage === "Error on connection" ||
            errorMessage === 'Error in server'||
            errorMessage === "Connection Timeout"
        ) { 
            return <Page400 />
        };

        return props.children;
    }

    const contextPayload = useMemo(() => (
        { setErrorMessage }), [setErrorMessage]
    );

    return (
        <ErrorHandlerContext.Provider value={contextPayload}>
            {renderContent()}
        </ErrorHandlerContext.Provider>
    )
}

const useErrorHandler = () => useContext(ErrorHandlerContext);

export { ErrorHandler, useErrorHandler }