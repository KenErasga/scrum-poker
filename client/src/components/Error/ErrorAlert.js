import React from 'react';
import Snackbar from '@material-ui/core/Snackbar';
import Alert from '@material-ui/lab/Alert';
import { makeStyles } from '@material-ui/core/styles';
import { useErrorHandler } from './ErrorHandler';

export const ErrorAlert = props => {
    const classes = useStyles();

    const { isError, errorMessage, setIsError, setErrorMessage } = useErrorHandler();

    const handleClose = () => {
        setIsError(false);
        setErrorMessage(undefined); 
    };

    return (
        <div className={classes.root}>
            <Snackbar open={isError} autoHideDuration={6000} onClose={handleClose}>
                <Alert severity="error" elevation={6} variant="filled" onClose={handleClose} {...props}>
                    {errorMessage}
                </Alert>
            </Snackbar>
        </div>
    )
};

const useStyles = makeStyles((theme) => ({
    root: {
        width: '100%',
        '& > * + *': {
            marginTop: theme.spacing(2),
        },
    },
}));