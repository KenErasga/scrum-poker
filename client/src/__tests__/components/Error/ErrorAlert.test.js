import React from 'react';
import renderer from 'react-test-renderer';
import {ErrorAlert} from '../../../components/Error/ErrorAlert';

jest.mock('../../../components/Error/ErrorHandler.js', () => ({
    useErrorHandler: () => ({
        isError: false,
        errorMessage: 'test error message',
        setIsError: jest.fn(),
        setErrorMessage: jest.fn()
    })
})); 

describe('ErrorAlert', () => {

    it('ErrorAlert renders correctly', () => {
    const tree = renderer
        .create(<ErrorAlert props='test' />)
        .toJSON();
    expect(tree).toMatchSnapshot();
    });

});
