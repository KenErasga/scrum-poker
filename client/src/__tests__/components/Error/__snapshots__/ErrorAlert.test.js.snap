// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorAlert ErrorAlert renders correctly 1`] = `
<div
  className="makeStyles-root-1"
/>
`;
